File: 94.BinaryTreeInorderTraversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> ret;
        traverse(ret, root);
        return ret;
    }
    
    void traverse(vector<int>& ret, TreeNode* node) {
        if(node == NULL) return;
        
        traverse(ret, node->left);
        ret.push_back(node->val);
        traverse(ret, node->right);
    }
};

File: 94.BinaryTreeInorderTraversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    
    
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> ret = new ArrayList<Integer>();
        traverse(ret, root);
        return ret;
    }
    
    void traverse(List<Integer> lt, TreeNode root) {
        if(root == null) return;
        
        traverse(lt, root.left);
        lt.add(root.val);
        traverse(lt, root.right);
    }
    
}